spec/unit/util/log/destinations_spec.rb

```diff
@@ -206,7 +206,17 @@ def expects_message_with_type(klass, level, eventlog_type, eventlog_id)
   it "supports the eventlog feature" do
     expect(Puppet.features.eventlog?).to be_truthy
   end
-
+  
+  it "should truncate extremely long log messages" do
+	long_msg = "x" * 32000
+	eventlog = stub('eventlog')
+	msg = Puppet::Util::Log.new(:level => :warning, :message => long_msg, :source => "a vogon ship")
```

Comment:
So I think you're on the right track here. 

Among the remaining things we need to do are
- create a new variable to hold the log message you expect _after_ the transformation
- set up an expectation that eventlog will receive the transformed message (as part of its entries)
- trigger the reporting - check out the previous tests that use a variable `dest` to accomplish this with the handle method